easybuild/toolchains/mpi/openmpi.py

```diff
@@ -57,6 +62,26 @@ class OpenMPI(Mpi):
 
     MPI_LINK_INFO_OPTION = '-showme:link'
 
+    def __init__(self, *args, **kwargs):
+        """Toolchain constructor"""
+        super(OpenMPI, self).__init__(*args, **kwargs)
+
+        self.orig_tmpdir = None
+
+    def prepare(self, *args, **kwargs):
+        """
+        Prepare for using OpenMPI library in toolchain environment
+        """
+        super(OpenMPI, self).prepare(*args, **kwargs)
+
+        # OpenMPI 2.x trips if path specified in $TMPDIR is too long
+        # see https://www.open-mpi.org/faq/?category=osx#startup-errors-with-open-mpi-2.0.x
+        self.orig_tmpdir = os.environ.get('TMPDIR')
+        ompi_ver = self.get_software_version(self.MPI_MODULE_NAME)[0]
+        if LooseVersion(ompi_ver) >= LooseVersion('2.0') and LooseVersion(ompi_ver) < LooseVersion('3.0'):
+            if len(self.orig_tmpdir) > 50:
+                env.setvar('TMPDIR', tempfile.mkdtemp(prefix='/tmp/'))
```

Comment:
are you really sure that's a good idea? I would alert the user and let them handle it?

Reply:
I don't see why this wouldn't be a good idea, since this is exactly the recommendation OpenMPI makes (set `$TMPDIR` to `/tmp`).
At least we're setting it to a unique subdirectory in `/tmp`...

Are you worried about the hardcoding of `/tmp`?